plugins/SpaceForNotes/main.lua

```diff
@@ -0,0 +1,36 @@
+local m = {
+    NORMAL = 0,
+    FORCE = 1,
+}
+
+-- Register all Toolbar actions and intialize all UI stuff
+function initUi()
+  app.registerUi{menu="Add one empty pages after all pdf page once", callback="AddEmpty", mode=m.NORMAL}
+  app.registerUi{menu="Add empty pages after every page", callback="AddEmpty", mode=m.FORCE}
+end
+
+function AddEmpty(mode)
+    local struct = app.getDocumentStructure()
+    for p=#struct.pages,1,-1 do
+        local page = struct.pages[p]
+        local pred = true
+        if mode ~= m.FORCE then
+            -- page p is a pdf page
+            pred = page.pdfBackgroundPageNo ~= 0
+            if pred then
+                -- check if page p+1 is also a pdf page (makes the operation idempotent)
+                local n = struct.pages[p+1]
+                -- no p is the last page -> also insert an empty one
+                if n == nil then
+                    pred = true
+                else
+                    pred = n.pdfBackgroundPageNo ~= 0
+                end
+            end
```

Comment:
Simpler:
```suggestion
            nextPage = struct.pages[p+1]
            pred = page.pdfBackgroundPageNo ~= 0 and (nextPage == nil or nextPage.pdfBackgroundPageNo ~= 0)
```

Reply:
Oh yes I see. Still maybe we should keep the comments (in some form) so that the intention can be reconstructed. (also `nextPage` should be `local`)